src/app/demo/demo.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit, ViewContainerRef } from '@angular/core';
+// import { User } from '../models/User';
+// import { Address } from '../models/address';
+import {NgbModal, ModalDismissReasons, NgbModalRef} from '@ng-bootstrap/ng-bootstrap';
+
+
+@Component({
+  selector: 'app-demo',
+  templateUrl: './demo.component.html',
+  styleUrls: ['./demo.component.scss']
+})
+export class DemoComponent implements OnInit {
+  unfreezeFlag = false;
+clicked= true;
+  failMsg: boolean;
+  successFlag = false;
+  private modalRef: NgbModalRef;
+  toggleButtonCondtion = false;
+  firstModel= true;
+  secondModel= false;
+  constructor(private modalService: NgbModal) {
+
+  }
+
+  ngOnInit() {
+  }
+
+
+  // tslint:disable-next-line:member-ordering
+  closeResult: string;
+
+  open(content) {
+    this.modalRef = this.modalService.open(content);
+    this.modalRef.result.then((result) => {
+      this.closeResult = `Closed with: ${result}`;
+    }, (reason) => {
+      this.closeResult = `Dismissed ${this.getDismissReason(reason)}`;
+    });
+  }
+
+
+  openLg(content) {
+    this.modalService.open(content, { size: 'lg' });
+  }
+
+  onNextClick() {
```

Comment:
When you use a single template for modal, you would be having a single method for freeze and unfreeze next button. 
based on freeze or unfreeze set the flag

Reply:
used only one function onNextClick() for both freeze and unfreeze function